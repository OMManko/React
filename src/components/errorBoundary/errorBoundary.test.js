import { shallow } from 'enzyme';
import React from 'react';
import ErrorBoundary from '../errorBoundary/errorBoundary';
import Header from "../layout/header/header";


describe('<ErrorBoundary />', () => {
    it('should render error boundary and match snapshot', () => {
        const component = shallow(<ErrorBoundary/>);
        expect(component).toMatchSnapshot();
    });

    it('should display error message if app throws an error', () => {
        const component = shallow(<ErrorBoundary>
            <Header/>
        </ErrorBoundary>);

        const error = new Error('test');

        component.find(Header).simulateError(error);
        expect(component.containsMatchingElement(<h2>Oops. Something went wrong.</h2>)).toEqual(true);
    });
});
