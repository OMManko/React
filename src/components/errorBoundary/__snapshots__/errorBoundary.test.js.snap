// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> should render error boundary and match snapshot 1`] = `ShallowWrapper {}`;
